tests/tcp.rs

```diff
@@ -571,6 +571,7 @@ fn connect_error() {
 
         for event in &events {
             if event.token() == Token(0) {
+                #[cfg(not(target_os = "hurd"))]
```

Comment:
What is causing this? This behaviour deviates from the expected.

Reply:
It's odd here. When I try a C equivalent of the test (connecting to a closed port, and expecting the resulting socket to be writable), I do see it writable. I don't know why I'm getting a non-writable event here. Actually I didn't find which part of the code sets `WRITABLE` in this case, I only saw `READABLE` in `src/sys/unix/waker/pipe.rs` for instance.

Reply:
> Actually I didn't find which part of the code sets `WRITABLE` in this case, I only saw `READABLE` in `src/sys/unix/waker/pipe.rs` for instance.

Mio doesn't set any events, is the OS/kernel that does that. The relevant check is in https://github.com/tokio-rs/mio/blob/1ce154579e1076260ea6934e89275912b01fae47/src/sys/unix/selector/poll.rs#L565.

Reply:
Ok, I see, `is_writable` is overloaded between the events and the interests.

So I dug more, what happens is that since tcp fast_open is not supported yet, but we set `SOCK_NONBLOCK`, after connect returned `EINPROGRESS` the socket notices the refused connection, and thus sets an error condition on the socket. `poll` thus returns `POLLERR` instead of `POLLOUT`, which indeed does make sense, I have added that as an alternative.

Reply:
Can we have a comment explaining this?

Reply:
Sure! Done so